modules/core/src/test/java/org/apache/ignite/internal/causality/IncrementalVersionedValueTest.java

```diff
@@ -308,33 +298,20 @@ void testConcurrentGetAndCompleteWithHistoryTrimming() throws Exception {
         // Set history size to 2.
         versionedValue.complete(3);
 
-        var barrier = new CyclicBarrier(2);
+        runRace(
+                () -> {
+                    versionedValue.update(4, (i, t) -> completedFuture(i + 1));
+                    // Trigger history trimming
+                    versionedValue.complete(4);
+                },
+                () -> {
+                    CompletableFuture<Integer> readerFuture = versionedValue.get(2);
 
-        CompletableFuture<Void> writerFuture = CompletableFuture.runAsync(() -> {
-            try {
-                barrier.await(1, TimeUnit.SECONDS);
-
-                versionedValue.update(4, (i, t) -> completedFuture(i + 1));
-                // Trigger history trimming
-                versionedValue.complete(4);
-            } catch (Exception e) {
-                throw new RuntimeException(e);
-            }
-        });
-
-        barrier.await(1, TimeUnit.SECONDS);
-
-        try {
-            CompletableFuture<Integer> readerFuture = versionedValue.get(2);
-
-            assertThat(readerFuture, willBe(1));
-        } catch (OutdatedTokenException ignored) {
-            // This is considered as a valid outcome.
-        }
+                    assertThat(readerFuture, willBe(1));
```

Comment:
It seems that you didn't include `OutdatedTokenException` handling this time. Why?

Reply:
Because I messed up, this test already failed and I've fixed it